adapters/alkimi/alkimi.go

```diff
@@ -0,0 +1,157 @@
+package alkimi
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/floors"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/prebid/openrtb/v19/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+const PRICE_MACRO = "${AUCTION_PRICE}"
+
+type AlkimiAdapter struct {
+	endpoint string
+}
+
+// Builder builds a new instance of the Alkimi adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	endpointURL, err := url.Parse(config.Endpoint)
+	if err != nil || len(endpointURL.String()) == 0 {
+		return nil, fmt.Errorf("invalid endpoint: %v", err)
+	}
+
+	bidder := &AlkimiAdapter{
+		endpoint: endpointURL.String(),
+	}
+	return bidder, nil
+}
+
+// MakeRequests creates Alkimi adapter requests
+func (adapter *AlkimiAdapter) MakeRequests(request *openrtb2.BidRequest, req *adapters.ExtraRequestInfo) (reqsBidder []*adapters.RequestData, errs []error) {
+	reqCopy := *request
+	reqCopy.Imp = _updateImps(reqCopy)
+	encoded, err := json.Marshal(reqCopy)
+	if err != nil {
+		errs = append(errs, err)
+	} else {
+		reqBidder := _buildBidderRequest(adapter, encoded)
+		reqsBidder = append(reqsBidder, reqBidder)
+	}
+	return
+}
+
+func _updateImps(bidRequest openrtb2.BidRequest) []openrtb2.Imp {
```

Comment:
@kalidas-alkimi any particular reason to add `_` as prefix to `updateImps()` and other methods? If intention was to make them private then should start methods with lowercase to make them private

Reply:
fixed